.github/workflows/images-branch.yml

```diff
@@ -18,9 +18,37 @@ on:  # yamllint disable-line rule:truthy
         description: PR number
 
 jobs:
+  approval:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set PR
+        run: echo "PR=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+        if: github.event_name != 'workflow_dispatch'
+
+      - name: Set PR (forks)
+        run: echo "PR=${{ inputs.pr }}" >> $GITHUB_ENV
+        if: github.event_name == 'workflow_dispatch'
+
+      # Skips push-updater-images by setting an output
+      - name: Check if pull request is approved
+        run: |
+          DECISION=$(gh pr view ${{ env.PR }} --json reviewDecision,state -t '{{.reviewDecision}}:{{.state}}')
+          echo "decision=$DECISION" >> $GITHUB_OUTPUT
+
+      # overwrite the previous result on pull request events on forks since forks can't publish to GHCR
+      - name: Skip forks
+        if: github.event.pull_request.head.repo.fork
+        run: echo "DECISION=FORK" >> $GITHUB_OUTPUT
```

Comment:
does it matter that DECISION is in caps here but it's lowercase when referenced?

Reply:
Fixed, nice catch!